jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/DeployServerlessApplicationSettings.kt

```diff
@@ -0,0 +1,49 @@
+// Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.services.lambda.deploy
+
+import com.intellij.openapi.components.PersistentStateComponent
+import com.intellij.openapi.components.State
+import com.intellij.openapi.module.Module
+import com.intellij.openapi.module.ModuleServiceManager
+
+@State(name = "deploySam")
+class DeployServerlessApplicationSettings : PersistentStateComponent<DeployServerlessApplicationConfiguration> {
```

Comment:
I'd love to make this a bit more generic for being able to store these sorts of settings across IDE restarts - is there a way we could create a pattern that would allow us to easily store other things, for example : bucket name for lambda update.

Reply:
As discussed offline, the bucket setting could be shared through all deployment actions in the same module. I've renamed this class to *DeploySettings* to allow them share settings like s3 bucket.